### PR TITLE
Remove support for sparse vectors.

### PR DESCRIPTION
**Pull request by jtibshirani**
Follow up to #48368. This PR removes support for `sparse_vector` on new
indices. On 7.x indices a `sparse_vector` can still be defined, but it is not
possible to index or search on it.

### CODE REVIEW

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@elasticmachine elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@elasticmachine elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@elasticmachine run @elasticmachine elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@mayya-sharipova this is now ready for a review (after some initial issues with test failures).

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

**Comment by jtibshirani** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

File: x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/SparseVectorFieldMapper.java [hunk 1]
```diff
@@ -8,44 +8,38 @@
...
 
-import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
-
 /**
  * A {@link FieldMapper} for indexing a sparse vector of floats.
+ *
+ * @deprecated The sparse_vector type was deprecated in 7.x and removed in 8.0. This mapper
+ * definition only exists so that 7.x indices can be read without error.
  */
+@Deprecated
```
Comment (mayya-sharipova) [inline x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/SparseVectorFieldMapper.java]: may be add `TODO: to remove this class in 9.0`

File: x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/mapper/SparseVectorFieldMapperTests.java [hunk 2]
```diff
@@ -71,197 +42,66 @@ protected boolean forbidPrivateIndexSettings() {
...
-                    .startObject("my-sparse-vector")
-                        .field(Integer.toString(-50), 100f)
-                    .endObject()
-                .endObject()),
-            XContentType.JSON));
-        });
-        assertThat(e.getCause(), instanceOf(IllegalArgumentException.class));
-        assertThat(e.getCause().getMessage(), containsString(
-            "dimension number must be a non-negative integer value not exceeding [65535], got [-50]"));
+    public void testSparseVectorWith7xIndex() throws Exception {
```
Comment (mayya-sharipova) [inline x-pack/plugin/vectors/src/test/java/org/elasticsearch/xpack/vectors/mapper/SparseVectorFieldMapperTests.java]: Nicely written test!

**Review by mayya-sharipova — APPROVED** [review]
@jtibshirani Thanks Julie, very nice approach for the bwc to keep just a definition of the type.

**Pull request merged**